app/forms/otp_verification_form.rb

```diff
@@ -0,0 +1,22 @@
+class OtpVerificationForm
+  def initialize(user, code)
+    @user = user
+    @code = code
+  end
+
+  def submit
+    @success = user.authenticate_direct_otp(code)
```

Comment:
what about putting this line in a method called `success` and returning  `{ success?: success }` from `submit` instead?  I think that would be easier to read in some ways...